lib/internal/fs/streams.js

```diff
@@ -64,41 +77,15 @@ function ReadStream(path, options) {
   this.closed = false;
 
   if (this.start !== undefined) {
-    if (!Number.isSafeInteger(this.start)) {
-      validateNumber(this.start, 'start');
-      if (!Number.isInteger(this.start))
-        throw new ERR_OUT_OF_RANGE('start', 'an integer', this.start);
-      throw new ERR_OUT_OF_RANGE(
-        'start',
-        '>= 0 and <= 2 ** 53 - 1',
-        this.start
-      );
-    }
-    if (this.start < 0) {
-      throw new ERR_OUT_OF_RANGE(
-        'start',
-        '>= 0 and <= 2 ** 53 - 1',
-        this.start
-      );
-    }
+    checkPosition(this.start, 'start');
 
     this.pos = this.start;
   }
 
   if (this.end === undefined) {
     this.end = Infinity;
   } else if (this.end !== Infinity) {
-    if (!Number.isSafeInteger(this.end)) {
-      if (typeof this.end !== 'number')
-        throw new ERR_INVALID_ARG_TYPE('end', 'number', this.end);
```

Comment:
It seems this error is no longer thrown.

Reply:
For example, if `this.end` is a string, then `this.end !== Infinity` is true, `!Number.isSafeInteger(this.end)` is true, `typeof this.end !== 'number'` is true, so the condition is hit and error is thrown.

Reply:
Oh, I used `validateNumber` instead here😅

which is from `internal/validators`:
https://github.com/nodejs/node/blob/c66c0732e0887c3f07a3e387d8a90b432eae134e/lib/internal/validators.js#L128-L131